### ISSUE TITLE
Shortcuts and Actions should allow detecting synthesized repeats (use new HardwareKeyboard API)

### ISSUE THREAD

**Issue by luigi-rosso** [flutter/flutter#95506]
This is regarding synthesized repeats generated by the platform when a key is held down.

There are specific cases where we want [Shortcuts to trigger Actions](https://docs.flutter.dev/development/ui/advanced/actions_and_shortcuts) on synthesized repeat, and specific other cases where we don't. It's important for the Shortcuts and Actions system to be able to discern when a keypress is a synthesized repeat.

Specifically, [ShortcutActivator.accepts](https://github.com/flutter/flutter/blob/c8981c5fbdc4d13673dbac3b88009d4a67f6dadd/packages/flutter/lib/src/widgets/shortcuts.dart#L212) currently receives a RawKeyEvent which is part of the old API. The new KeyEvent from HardwareKeyboard allows us to distinguish if the keypress is a synthesized repeat.

It seems like migrating [Actions and Shortcuts](https://docs.flutter.dev/development/ui/advanced/actions_and_shortcuts) to the newer [HardwareKeyboard](https://github.com/flutter/flutter/pull/83752) API would allow us to do this.

**Comment by luigi-rosso** [comment]
Example use case: 

User holds Command + G to group a selection.
If they keep Command + G held, the selection will keep getting grouped into new groups until the user releases Command + G because the Shortcut will keep triggering the Group action while the synthesized repeats are fired. The user perceives this as a bug.

Instead, if the system could detect the keypress is a synthesized repeat (which the new HardwareKeyboard API supports) we could return false in the ShortcutActivator.accepts method when a non-repeating action finds a synthesized repeat event.

In summary, some shortcuts should trigger actions on synthesized repeat, some should not, we would like to have control over this. We currently do this by using our own Shortcut/Action model on top of our own channel sending keypresses (other issue).

I'll cook up an example to tack on to this issue too shortly.

**Comment by luigi-rosso** [comment]
Here's a small repro example here:
https://github.com/luigi-rosso/flutter-shortcuts-actions-bug-95506

The example fires a "GroupIntent" when the "G" key is pressed. The counter increments whenever a GroupAction is fired. Notice that it keeps repeating when the key is pressed as the system is generating synthesized key repeats (what the new HardwareKeyboard API discerns as KeyRepeatEvent).

The crux of it all boils down to this:
https://github.com/luigi-rosso/flutter-shortcuts-actions-bug-95506/blob/8fd622eac7ed97b8a61ecbcf026505ff45541fec/lib/main.dart#L105-L108

**Comment by csells** [comment]
cc @HansMuller @cbracken this is an important issue for Rive.

**Comment by csells** [comment]
@gspencergoog thoughts about what Rive might do to work around this issue?

**Comment by gspencergoog** [comment]
@dkwingsmt Can we switch the Framework over to using the new keyboard KeyEvent yet? That would be the more straightforward solution to this problem (although perhaps not the simplest, given it might still need a bunch of work).

**Comment by dkwingsmt** [comment]
Yes, we definitely want to resolve it by switching. I think it's about time we switch too, probably after https://github.com/flutter/flutter/issues/88021 though.
